File: src/test/java/mnm/mods/tabbychat/test/ChatTextUtilsTest.java
package mnm.mods.tabbychat.test;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

import mnm.mods.tabbychat.util.ChatTextUtils;
import net.minecraft.util.text.ITextComponent;
import net.minecraft.util.text.TextComponentString;
import net.minecraft.util.text.TextFormatting;
import net.minecraft.util.text.event.ClickEvent;

public class ChatTextUtilsTest {

    @Test
    public void testSubChat() {
        assertEquals(makeChat(false), ChatTextUtils.subChat(makeChat(true), 7));
    }

    private static ITextComponent makeChat(boolean tag) {

        ITextComponent chat = new TextComponentString(tag ? "[test] " : "");
        chat.getStyle().setBold(true);
        {
            ITextComponent colored = new TextComponentString("This should be green. ");
            colored.getStyle().setColor(TextFormatting.GREEN);
            chat.appendSibling(colored);
        }
        chat.appendText(" ");
        {
            ITextComponent link = new TextComponentString("This is a link.");
            link.getStyle().setClickEvent(new ClickEvent(ClickEvent.Action.OPEN_URL, "http://google.com/"));
            chat.appendSibling(link);
        }
        return chat;
    }
}

File: src/test/kotlin/mnm/mods/tabbychat/test/ChatTextUtilsTest.kt
package mnm.mods.tabbychat.test

import mnm.mods.tabbychat.util.ChatTextUtils
import net.minecraft.util.text.ITextComponent
import net.minecraft.util.text.StringTextComponent
import net.minecraft.util.text.TextFormatting
import net.minecraft.util.text.event.ClickEvent
import org.junit.Assert.assertEquals
import org.junit.Test

class ChatTextUtilsTest {

    @Test
    fun testSubChat() {
        assertEquals(makeChat(false), ChatTextUtils.subChat(makeChat(true), 7))
    }

    private fun makeChat(tag: Boolean): ITextComponent {

        return StringTextComponent(if (tag) "[test] " else "").applyTextStyle(TextFormatting.BOLD)
                .appendSibling(StringTextComponent("This should be green")
                        .applyTextStyle(TextFormatting.GREEN))
                .appendText(" ")
                .appendSibling(StringTextComponent("This is a link.").applyTextStyle {
                    it.clickEvent = ClickEvent(ClickEvent.Action.OPEN_URL, "http://google.com")
                })

    }
}
